### ISSUE TITLE
Bug in test_S3DIS.py reshape

### ISSUE THREAD

**Issue by SimonBXY** [mellody11/RandLA-Net-Pytorch-New#20]
In line 142, stacked_probs = torch.reshape(stacked_probs, [cfg.val_batch_size, cfg.num_points, cfg.num_classes]).
You actually want to TRANSPOSE the last 2 dims.
What your code does is like a filename change, which has no effect on the file content itself.
This would lead to predictions all being wrong!
Please fix this to transpose(1, 2)!

**Comment by liuxuexun** [comment]
Hi, thank you for your attention to this repo.

I trained a new model on area5 and ran test_S3DIS.py, but did not reproduce the error.

Here is my log:
```
Epoch   0, end. Min possibility = 7.7

Confusion on sub clouds
62.41 | 91.65 97.36 79.54  0.00 26.85 55.87 50.36 75.68 84.13 64.40 69.70 62.55 53.27 


Reproject Vote #7
Confusion on full clouds
Area_5_office_16 Acc:0.9122831478750503
Area_5_storage_1 Acc:0.5130369722037919
Area_5_storage_4 Acc:0.7921005077015747
Area_5_office_11 Acc:0.9478143548513356
Area_5_office_19 Acc:0.8454949296560598
Area_5_office_23 Acc:0.8810566942806416
Area_5_office_30 Acc:0.9097329594862993
Area_5_office_26 Acc:0.8122493248213394
Area_5_hallway_13 Acc:0.8901910049549174
Area_5_storage_3 Acc:0.7082341145506956
Area_5_office_12 Acc:0.8754266150999549
Area_5_office_34 Acc:0.8073912603795633
Area_5_office_6 Acc:0.892900537175989
Area_5_office_10 Acc:0.8894103667313973
Area_5_WC_2 Acc:0.8994111675126903
Area_5_office_17 Acc:0.8323647945532775
Area_5_office_2 Acc:0.9269862830470638
Area_5_hallway_6 Acc:0.9302430565963357
Area_5_storage_2 Acc:0.5569582941923544
Area_5_office_14 Acc:0.8584262483144485
Area_5_office_3 Acc:0.8831668119537763
Area_5_hallway_4 Acc:0.9363313743086797
Area_5_office_7 Acc:0.884283978661914
Area_5_conferenceRoom_3 Acc:0.8936268731272045
Area_5_office_5 Acc:0.9544094680300031
Area_5_office_20 Acc:0.8411619042427687
Area_5_office_24 Acc:0.7476403931461257
Area_5_office_27 Acc:0.873459849975656
Area_5_office_21 Acc:0.8978216545267635
Area_5_office_40 Acc:0.8757254659689052
Area_5_conferenceRoom_2 Acc:0.911616312682816
Area_5_office_41 Acc:0.8488677652637152
Area_5_hallway_11 Acc:0.9808695973584348
Area_5_hallway_2 Acc:0.9333658556833364
Area_5_office_32 Acc:0.8623373542798124
Area_5_office_22 Acc:0.9051324864067781
Area_5_office_4 Acc:0.9209236503790238
Area_5_office_1 Acc:0.8250965525353617
Area_5_hallway_15 Acc:0.8582102721713112
Area_5_hallway_10 Acc:0.9806206417925245
Area_5_office_9 Acc:0.8794405494036551
Area_5_lobby_1 Acc:0.9631430571585341
Area_5_pantry_1 Acc:0.7045354670836573
Area_5_office_28 Acc:0.881585991753193
Area_5_office_29 Acc:0.776094013528953
Area_5_office_35 Acc:0.9387375713488998
Area_5_office_25 Acc:0.8372599425851827
Area_5_office_36 Acc:0.9014472402597402
Area_5_WC_1 Acc:0.830132286459405
Area_5_office_37 Acc:0.8125526619269314
Area_5_office_15 Acc:0.8802196726461721
Area_5_office_13 Acc:0.8541473671970387
Area_5_office_33 Acc:0.8832729827722314
Area_5_office_38 Acc:0.7948333131832132
Area_5_office_42 Acc:0.9090834811564067
Area_5_hallway_12 Acc:0.9872065519046349
Area_5_office_31 Acc:0.820102993862661
Area_5_hallway_14 Acc:0.9131558608267862
Area_5_office_8 Acc:0.9291081043204659
Area_5_hallway_5 Acc:0.9460559292025515
Area_5_hallway_9 Acc:0.9528271580278971
Area_5_hallway_7 Acc:0.9326114534107314
Area_5_office_18 Acc:0.9188740092421146
Area_5_hallway_8 Acc:0.9517504394607257
Area_5_office_39 Acc:0.8393140176933215
Area_5_conferenceRoom_1 Acc:0.8918814686402802
Area_5_hallway_1 Acc:0.7308046922904593
Area_5_hallway_3 Acc:0.9642941913416362
--------------------------------------------------------------------------------------
63.08 | 92.13 97.39 79.91  0.00 27.43 58.53 50.40 76.61 85.59 65.38 69.97 63.04 53.71 
--------------------------------------------------------------------------------------
```

Could you provide your log? Have you made any modification of the code?

**Comment by liuxuexun** [comment]
The `reshape` operation is correct, because the shape of the original tensor is `[B*N, C]`, and we want a `[B, N, C]` tensor for voting (line 148~152).

I hope this explaination may help.

**Comment by SimonBXY** [comment]
I'm very sorry. I now see that in compute_loss logits are transposed and saved as valid logits.
I modified some code quite a while ago and in my version logits from model output is used instead of valid_logits, which caused the bug.
I apologizie again for my mistake. And thank you for your prompt reply and patience!
